### ISSUE TITLE
Undocumented rule for adding spaces before inline comments?

### ISSUE THREAD

**Issue by ghost** [swiftlang/swift-format#525]
In our codebase, the linter complains about hundreds of instances such as:

```swift
title: "Settings", // TODO: ....
```

and

```swift
CustomSection(title: "App Info") { // TODO: ...
```

It is always about having only 1 space before the inline comment forward slashes (`//`). The linter wants to have 2 spaces, as in

```swift
title: "Settings",  // TODO: ....
```

Why ? And what undocumented rule is enforcing that? And why is it undocumented?

**Comment by rikner** [comment]
I also think there should be a configurable rule for this in order to be able to align this with other formatters in multi-language codebases.

**Comment by ephemer** [comment]
Agreed, to me the current behaviour looks like an error. Why randomly enforce two spaces?

**Comment by allevato** [comment]
I'm happy to review a PR that adds a configuration setting to change the number of spaces enforced before an end-of-line comment if someone wants to implement it. I think it should be a fairly straightforward change to `TokenStreamCreator`.

**Comment by ahoppen** [comment]
Tracked in Apple’s issue tracker as rdar://126948345

**Comment by ZachNagengast** [comment]
For anyone finding this before it gets released, @dduan added a PR to make this configurable  https://github.com/swiftlang/swift-format/pull/776 with `"spacesBeforeEndOfLineComments": 1,`🙏  you can enable it by installing the HEAD version of swift-format, here's an example using Homebrew
```
brew uninstall swift-format
brew install swift-format --HEAD
```

Previously it was hardcoded to `2` https://github.com/swiftlang/swift-format/pull/776/files#diff-c6c2e2a43fac11c6c1cda074a1e1b01693672cc219a2259994b2630c15e628dfL3174

**Comment by allevato** [comment]
Closing as fixed by #776.
